pkg/detector/detector.go

```diff
@@ -1157,15 +1170,15 @@ func (d *ResourceDetector) HandleClusterPropagationPolicyDeletion(policyID strin
 			// ClusterResourceBinding, otherwise might lose the chance to do that in a retry loop (in particular, the
 			// marks was successfully removed from ClusterResourceBinding, but resource template not), since the
 			// ClusterResourceBinding will not be listed again.
-			if err := d.CleanupResourceTemplateMarks(binding.Spec.Resource, cleanupMarksFun); err != nil {
+			if err := d.CleanupResourceTemplateMarks(binding.Spec.Resource, policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel, policyID, cleanupMarksFun); err != nil {
```

Comment:
Can we generate the list of labels/annotations that need to be removed? The list can be generated inside of the `CleanupResourceTemplateMarks` or passed into it.

Reply:
Do you mean
https://github.com/karmada-io/karmada/blob/920ac8931e7aeccae87b7bc123c2b001060dcda0/pkg/detector/detector.go#L66-L78

Reply:
@RainbowMango How about now?

Reply:
You can review the code first, and I'll verify the functionality locally tomorrow